janitor/functions.py

```diff
@@ -2630,3 +2630,51 @@ def to_datetime(df: pd.DataFrame, column: str, **kwargs) -> pd.DataFrame:
     df[column] = pd.to_datetime(df[column], **kwargs)
 
     return df
+
+
+@pf.register_dataframe_method
+def groupby_agg(
+        df: pd.DataFrame,
+        by: str,
+        new_column: str,
+        agg_column: str,
+        agg: Union[Callable, str, List, Dict],
+        axis: int=0
+) -> pd.DataFrame:
+    """
+
+    Allow one to chain a groupby and a merge
+
+    Functional usage example:
```

Comment:
```suggestion
    Without this function, we would have to break out of method chaining:

    .. code-block:: python

        df_grp = df.groupby(...).agg(...)
        df = df.merge(df_grp, ...)

    Now, this function can be method-chained:
```

Reply:
I can add this, or just remove completely if you want people to move away?

Reply:
This part was added so as to contrast how without your contrib, we'd have to break out of method chaining. I would love for this part to be kept, so that it's evident for a new person that this function makes life much easier!

Reply:
Added the text you suggested